xonsh/tools.py

```diff
@@ -595,19 +584,16 @@ def ensure_int_or_slice(x):
 
 def is_string_set(x):
     """Tests if something is a set"""
-    if isinstance(x, string_types):
-        return False
-    else:
-        return (isinstance(x, Set) and
-                all([isinstance(a, string_types) for a in x]))
+    return (isinstance(x, Set) and
+            all(isinstance(a, str) for a in x))
 
 
 def csv_to_set(x):
     """Convert a comma-separated list of strings to a set of strings."""
     if not x:
         return set()
     else:
-        return set(x.split(','))
+        return set(y.strip() for y in x.split(','))
```

Comment:
> it's used for user-entered data who tend to put an extra space after a comma.

sure, which is fine, if you want to give it a new name and change the ensurer to be something more useful. My problem is only that it is named CSV but you are proposing to change it to something that is not CSV.  So either the name can change or it should remain CSV compliant.  I don't personally mind too much of either of these :).

> oh, and are sets used here with the rational for uniqueness too? i would porpose another approach to this with the mentioned ensurer enhancement.

IIRC the sets were for uniqueness here. Please feel free to revert this change and address any improvements in another PR.  No reason this all has to be done in this one. I think it slows this one down :)